### ISSUE TITLE
Add support for Outbound Rules in Standard Load Balancer

### ISSUE THREAD

**Issue by PleaseStopAsking** [terraform-providers/terraform-provider-azurerm#1984]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure has just announced support for Outbound Rules on Standard Sku Load Balancers. Outbound rules make it simple to configure public Standard Load Balancer's outbound network address translation. You have full declarative control over outbound connectivity to scale and tune this ability to your specific needs.

### New or Affected Resource(s)

azurerm_lb_outbound_rule

### Potential Terraform Configuration

```hcl
resource "azurerm_public_ip" "test" {
  name                         = "PublicIPForLB"
  location                     = "West US"
  resource_group_name          = "${azurerm_resource_group.test.name}"
  public_ip_address_allocation = "static"
}

resource "azurerm_lb" "test" {
  name                = "TestLoadBalancer"
  location            = "West US"
  resource_group_name = "${azurerm_resource_group.test.name}"

  frontend_ip_configuration {
    name                 = "PublicIPAddress"
    public_ip_address_id = "${azurerm_public_ip.test.id}"
  }
}

resource "azurerm_lb_backend_address_pool" "test" {
  resource_group_name = "${azurerm_resource_group.test.name}"
  loadbalancer_id     = "${azurerm_lb.test.id}"
  name                = "BackEndAddressPool"
}

resource "azurerm_lb_outbound_rule" "test" {
  resource_group_name = "${azurerm_resource_group.test.name}"
  loadbalancer_id = "${azurerm_lb.test.id}"
  name = "outboundRule"
  frontend_ip_configurations = "PublicIPAddress"
  idleTimeoutInMinutes = 60
  enableTcpReset = $true 
  protocol = "All"
  backendAddressPool = ["${azurerm_lb_backend_address_pool.test.id}"]
}
```

### References

https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-rules-overview
https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-outbound-connections#ilpip

**Comment by metacpp** [comment]
@PleaseStopAsking thanks for opening this request.

You can find all released packages for `network` at:
https://github.com/Azure/azure-sdk-for-go/blob/master/services/network/mgmt

For `Outbound Rules`, you can find it at `2018-08-01` API:
https://github.com/Azure/azure-sdk-for-go/blob/master/services/network/mgmt/2018-08-01/network/loadbalanceroutboundrules.go

**Comment by PleaseStopAsking** [comment]
@metacpp As Outbound Rules are supported in the SDK starting at 2018-08-01, would this require updating the SDK in /vendor to the new version for all network resources or is there a way to only use it for load balancer resources?

I tested simply adding the new version to the network dir but as expected there are other dependencies with the new version.

**Comment by metacpp** [comment]
> @metacpp As Outbound Rules are supported in the SDK starting at 2018-08-01, would this require updating the SDK in /vendor to the new version for all network resources or is there a way to only use it for load balancer resources?

We're using `v21.3.0` for Azure Go SDK, which contains the `2018-08-01`:
https://github.com/Azure/azure-sdk-for-go/tree/v21.3.0/services/network/mgmt

You can just use below command to fetch package for `2018-08-01`:
```
govendor fetch "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-08-01/network"@v21.3.0
```

**Comment by mcharriere** [comment]
Hi,
I could start working on this new resource now that it's using v24 for the SDK. 
Is there anything that I should check before adding this feature?

**Comment by djsly** [comment]
@mcharriere are you picking this up in the end?

**Comment by mcharriere** [comment]
> @mcharriere are you picking this up in the end?

Yes, I'm about to finish it. just polishing it a bit. I hope I'll have it done by Mondey

**Comment by djsly** [comment]
Super! Good news.

**Comment by mcharriere** [comment]
Hi. I think that this could be closed after merging the pull request. Did I miss something? Thanks

**Comment by tombuildsstuff** [comment]
@mcharriere thanks for the ping - agreed this can now be closed since #2912 has shipped - thanks 👍
